apps/settings/js/panels/homescreens_list/homescreens_list.js

```diff
@@ -0,0 +1,181 @@
+  var SettingsService = require('modules/settings_service');
+  var SettingsCache = require('modules/settings_cache');
+  var ManifestHelper = require('shared/manifest_helper');
+
+  var HomescreensList = function ctor_homescreens_list() {
+    this._apps = [];
+    this.elements = {};
+    this.manifestURL = '';
+  };
+
+  HomescreensList.prototype = {
+    /**
+     * initialization.
+     *
+     * @param {Array.<Object>} elements An array of HTML elements.
+     */
+    init: function hl_init(elements) {
+      this.elements = elements;
+      this._renderHomescreens();
+
+      window.addEventListener('applicationinstall',
+        this._renderHomescreens.bind(this));
+      window.addEventListener('applicationuninstall',
+        this._renderHomescreens.bind(this));
+
+      this.elements.homescreensList.addEventListener('click',
+        this._handleListClick.bind(this));
+      this.elements.moreLink.addEventListener('click',
+        this._showMoreHomescreens.bind(this));
+
+      // Refresh the list whenever apps are installed/uninstalled.
+      AppsCache.addEventListener('oninstall',
+        this._renderHomescreens.bind(this));
+      AppsCache.addEventListener('onuninstall',
+        this._renderHomescreens.bind(this));
+
+      SettingsListener.observe('homescreen.manifestURL', '', manifestURL => {
+        this.manifestURL = manifestURL;
+        // There is a race condition, so we need to refresh the list.
+        this.listBuilder();
+      });
+    },
+
+    /**
+     * handle list click event
+     * @private
+     */
+    _handleListClick: function hl_handleListClick(evt) {
+      var target = evt.target;
+      var index = target.dataset.appIndex;
+
+      while (index === undefined) {
+        target = target.parentNode;
+        index = target.dataset.appIndex;
+      }
+
+      var app = this._apps[index];
+      const settings = window.navigator.mozSettings;
+
+      if (evt.target.nodeName === 'GAIA-RADIO') {
+        // Change the current home screen.
+        settings.createLock().set({
+          'homescreen.manifestURL': app.manifestURL
+        });
+        return;
+      }
+
+      // Otherwise, open the home screen detail panel.
+      var manifest = new ManifestHelper(app.manifest || app.updateManifest);
+
+      SettingsService.navigate('homescreen-details', {
+        name: manifest.name,
+        author: manifest.developer,
+        version: manifest.version,
+        description: manifest.description,
+        removable: app.removable,
+        app: app
+      });
+    },
+
+    /**
+     * Build list item.
+     *
+     * @param {Object} app   app object.
+     * @param {Number} index link index.
+     * @private
+     */
+    _listItemBuilder: function hl_listItemBuilder(app, index) {
+      var item = document.createElement('li');
+      item.className = 'pack-split';
+      item.dataset.appIndex = index;
+
+      var radioLabel = document.createElement('label');
```

Comment:
To make it more readable, please add a comment sample template above to demonstrate how the generated code looks like, ex:

```
<li class="pack-split" data-appIndex=1>
  <gaia-switch>
    <label></label>
  </gaia-switch>
</li>
```